projects/igniteui-angular/src/lib/date-range/igx-date-range.component.ts

```diff
@@ -0,0 +1,696 @@
+    }
+
+    private get dialogOverlaySettings(): OverlaySettings {
+        return Object.assign({}, this._dialogOverlaySettings, this.overlaySettings);
+    }
+
+    private _value: DateRange;
+    private _inputFormat: string;
+    private $destroy = new Subject();
+    private $toggleClickNotifier = new Subject();
+    private _onChangeCallback: (...args: any[]) => void;
+    private _positionSettings: PositionSettings;
+    private _dialogOverlaySettings: OverlaySettings = {
+        closeOnOutsideClick: true,
+        modal: true,
+        positionStrategy: new GlobalPositionStrategy()
+    };
+    private _dropDownOverlaySettings: OverlaySettings = {
+        closeOnOutsideClick: true,
+        modal: false
+    };
+
+    constructor(public element: ElementRef,
+        @Inject(LOCALE_ID) private _locale: any,
+        @Optional() @Inject(DisplayDensityToken) protected _displayDensityOptions: IDisplayDensityOptions) {
+        super(_displayDensityOptions);
+        this._onChangeCallback = (...args: any[]) => { };
+        this.locale = this.locale || this._locale;
+    }
+
+    /**
+     * Opens the date range picker's dropdown or dialog.
+     *
+     * @example
+     * ```html
+     * <igx-date-range-picker #dateRange></igx-date-range-picker>
+     *
+     * <button (click)="dateRange.open()">Open Dialog</button
+     * ```
+     */
+    public open(overlaySettings?: OverlaySettings): void {
+        const range: Date[] = [];
+        if (this.value) {
+            if (this.value.start) {
+                range.push(this.value.start);
+                if (this.value.end) {
+                    range.push(this.value.end);
+                }
+            } else {
+                this.value = { start: null, end: null };
+            }
+        }
+        if (range.length > 0) {
+            this.value = { start: range[0], end: range[range.length - 1] };
+        }
+
+        this.updateCalendar();
+        const settings = this.mode === InteractionMode.Dialog ? this.dialogOverlaySettings : this.dropdownOverlaySettings;
+        if (this.toggleDirective.collapsed) {
+            this.toggleDirective.open(Object.assign(settings, overlaySettings));
+        }
+    }
+
+    /**
+     * Closes the date range picker's dropdown or dialog.
+     *
+     * @example
+     * html```
+     * <igx-date-range-picker #dateRange></igx-date-range-picker>
+     *
+     * <button (click)="dateRange.close()">Close Dialog</button>
+     * ```
+     */
+    public close(): void {
+        if (!this.toggleDirective.collapsed) {
+            this.toggleDirective.close();
+        }
+    }
+
+    /**
+     * Toggles the date range picker's dropdown or dialog
+     *
+     * @example
+     * html```
+     * <igx-date-range-picker #dateRange></igx-date-range-picker>
+     *
+     * <button (click)="dateRange.toggle()">Toggle Dialog</button>
+     * ```
+     */
+    public toggle(): void {
```

Comment:
as we provide `OverlaySettings` to `open` method we should provide them and to `toggle` here